smart-py/marketplace.py

```diff
@@ -53,12 +59,70 @@ def cancel_swap(self, params):
     def update_fee(self, params):
         sp.verify(sp.sender == self.data.manager)
         self.data.fee = params
-        
+
     @sp.entry_point
-    def update_manager(self, params):
+    def update_fee_account(self, params):
         sp.verify(sp.sender == self.data.manager)
+        self.data.fee_account= params
+
+    @sp.entry_point
+    def update_manager(self, params):
+        sp.if sp.now - self.data.last_ping > 100 * 24 * 3600:
```

Comment:
This is from the Discord thread: "Can you explain the thinking of the delay for the secondary account to be allowed to update the manager? If there was a bus incident, might it not become urgent to be able to change the contract manager? Also, would it make sense to create a PR for a code review to add comments there?"

jagracar: "I don't see any other solution than the time delay to take control of the manager account. The delay could be reduced to some weeks, but that means that the manager will have to ping quite often the contract to keep control. All that will need to be agreed if it's implemented."

Reply:
We could even add another entry point to change the delay.

Reply:
i think a long delay is ok, things can keep running as is until the timer is up